include/highfive/bits/H5Converter_misc.hpp

```diff
@@ -267,32 +267,126 @@ struct data_converter<boost::numeric::ublas::matrix<T>, void> {
         assert(_dims.size() == 2);
     }
 
     inline typename type_of_array<T>::type* transform_read(MatrixTMN& array) {
-        if (_dims[0] != M || _dims[1] != N) {
-            array.resize(_dims[0], _dims[1]);
+
+        if (_dims[0] != (size_t)array.rows() || _dims[1] != (size_t)array.cols()) {
+            array.resize((Eigen::Index)_dims[0], (Eigen::Index)_dims[1]);
         }
-        return array.data;
+        return array.data();
     }
 
     inline typename type_of_array<T>::type* transform_write(MatrixTMN& array) {
-        return array.data;
+        return array.data();
     }
 
     inline void process_result(MatrixTMN&) {}
 
     std::vector<size_t> _dims;
 };
+
+template <typename T, int M, int N>
+inline void vectors_to_single_buffer(const std::vector<Eigen::Matrix<T,M,N>>& vec,
+                                     const std::vector<size_t>& dims,
+                                     const size_t current_dim,
+                                     std::vector<T>& buffer) {
+
+    check_dimensions_vector(compute_total_row_size(vec), dims[current_dim], current_dim);
+    for (const auto& k : vec) {
+        std::copy(k.data(), k.data()+k.size(), std::back_inserter(buffer));
+    }
+}
+
+// apply conversion to std::vector of eigen matrix
+template <typename T, int M, int N>
+struct data_converter<std::vector<Eigen::Matrix<T,M,N>>, void> {
+
+    typedef typename Eigen::Matrix<T, M, N> MatrixTMN;
+
+    inline data_converter(const std::vector<MatrixTMN>& , DataSpace& space)
+        : _dims(space.getDimensions()), _space(space) {
+        assert(_dims.size() == 2);
+    }
+
+    inline typename type_of_array<T>::type*
+    transform_read(std::vector<MatrixTMN>& vec) {
+        _vec_align.resize(compute_total_size(_space.getDimensions()));
+        return _vec_align.data();
+    }
+
+    inline typename type_of_array<T>::type*
+    transform_write(std::vector<MatrixTMN>& vec) {
+        _vec_align.reserve(compute_total_size(vec));
+        vectors_to_single_buffer<T, M, N>(vec, _dims, 0, _vec_align);
+        return _vec_align.data();
+    }
+
+    inline void process_result(std::vector<MatrixTMN>& vec) {
+        T* start = _vec_align.data();
+        if(vec.size() > 0) {
+            for(auto& v : vec){
+                v = Eigen::Map<MatrixTMN>(start, v.rows(), v.cols());
+                start += v.rows()*v.cols();
+            }
+        }
+        else {
+            if(M == -1 || N == -1) {
+                std::ostringstream ss;
+                ss << "Dynamic size (-1) used without pre-defined data layout for vector elements.\n";
+                ss << "Initiliaze vector elements using Zero, i.e. vector<MatrixXd> vec(5, MatrixXd::Zero(20,5))";
+                throw DataSetException(ss.str());
+            }
+            else
+            {
+                for (size_t i = 0; i < _dims[0] / M; ++i) {
+                    vec.emplace_back(Eigen::Map<MatrixTMN>(start, M, N));
```

Comment:
Probably:
```
vec.emplace_back(start, M, N);
```
but can't be certain from GitHub UI

Reply:
That constructor does not work for dynamic sized Eigen::Matrix (i.e. M = -1). It was the only thing I could find that works. Please let me know if you know any other way.